extensions/content-manager/admin/src/components/SettingsViewWrapper/index.js

```diff
@@ -0,0 +1,329 @@
+    actions: getPluginHeaderActions(),
+    title: {
+      label: formatMessage(
+        {
+          id: `${pluginId}.components.SettingsViewWrapper.pluginHeader.title`,
+        },
+        { name: upperFirst(name) }
+      ),
+    },
+    content: formatMessage({
+      id: `${pluginId}.components.SettingsViewWrapper.pluginHeader.description.${
+        isEditSettings ? 'edit' : 'list'
+      }-settings`,
+    }),
+  };
+
+  const getRelationNameOptions = () => {
+    const attributes = getAttributes;
+    const relationName = initialData.settings.relationName
+    const options = Object.keys(attributes).filter(attr => {
+      const type = get(attributes, [attr, 'type'], '');
+      const relationType  = get(attributes, [attr, 'relationType'], '');
+      if(type === 'relation' && relationType === 'oneWay' ) {
+        return attr
+      }
+    });
+    if(!attributes[relationName]) {
+      return ['', ...options];
+    }
+    return options;
+  }
+
+  const getSelectOptions = input => {
+    if (input.name === 'settings.defaultSortBy' || input.name === 'settings.relationKey') {
+      return [
+        'id',
+        ...getListDisplayedFields().filter(
+          name =>
+            get(getAttributes, [name, 'type'], '') !== 'media' &&
+            name !== 'id' &&
+            get(getAttributes, [name, 'type'], '') !== 'richtext'
+        ),
+      ];
+    }
+
+    if (input.name === 'settings.mainField') {
+      const attributes = getAttributes;
+      const options = Object.keys(attributes).filter(attr => {
+        const type = get(attributes, [attr, 'type'], '');
+
+        return (
+          ![
+            'json',
+            'text',
+            'relation',
+            'component',
+            'boolean',
+            'date',
+            'media',
+            'richtext',
+          ].includes(type) && !!type
+        );
+      });
+      return options;
+    }
+
+    if (input.name === 'settings.relationName') {
+      const options =  getRelationNameOptions()
+      return options;
+    }
+
+    return input.options;
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    toggleWarningSubmit();
+    emitEvent('willSaveContentTypeLayout');
+  };
+
+  if (isLoading) {
+    return <LoadingIndicatorPage />;
+  }
+
+  const filterInputsByRelation = () => {
+    const filteredInputs = inputs.filter(input => {
+      return input.name !== 'settings.relationKey' && input.name !== 'settings.relationName'
+    });
+    const relationNameOptions = getRelationNameOptions();
+    if(modifiedData.settings && relationNameOptions.length) {
+      return modifiedData.settings.relation
+        ? inputs 
+        : filteredInputs
+    }
+    return filteredInputs;
+  }
+
+  const generateInputValue = (input) => {
+    if(input.name === 'settings.relation'){
+      const relationNameOptions = getRelationNameOptions();
+      return relationNameOptions.length > 0 ? get(modifiedData, input.name, '') : false;
```

Comment:
false? You using it to pass value props, maybe instead of false return null or undefined or empty string as you do it as fallback in get function?

Reply:
it is value for switch so I think that true/false is proper value

Reply:
ok, so maybe change default value of lodash get function to true/false?